sdk/logical/logical.go

```diff
@@ -131,6 +131,28 @@ type Paths struct {
 	// should be seal wrapped with extra encryption. It is exact matching
 	// unless it ends with '/' in which case it will be treated as a prefix.
 	SealWrapStorage []string
+
+	// WriteForwardedStorage are storage paths that, when running on a PR
+	// Secondary cluster, cause a GRPC call up to the PR Primary cluster's
+	// active node to handle storage.Put(...) and storage.Delete(...) events.
+	// These paths MUST include a {{clusterId}} literal, which the write layer
+	// will resolve to this cluster's UUID ("replication set" identifier).
+	// storage.List(...) and storage.Get(...) operations occur from the
+	// locally replicated data set, but can use path template expansion to be
+	// identifier agnostic.
+	//
+	// These paths require careful considerations by developers to use. In
+	// particular, writes on secondary clusters will not appear (when a
+	// corresponding read is issued immediately after a write) until the
+	// replication from primary->secondary has occurred. This replication
+	// triggers an InvalidateKey(...) call on the secondary, which can be
+	// used to detect the write has finished syncing. However, this will
+	// likely occur after the request has finished, so it is important to
+	// not block on this occurring.
+	//
+	// On standby nodes, like all storage write operations, this will trigger
+	// an ErrReadOnly return.
```

Comment:
You state here that on standby nodes it'll utilize the existing read only error, but below you're wrapping the barrier with the forwarding backend. The standby read-only storage implementation lives below the barrier. I worry that might make this statement inaccurate. Can you test that?

Reply:
@briankassouf In retrospect, I think I wrote this confusing two layers:

 - The localStoragePaths layer, which sits above us (between us and the plugin), returning ErrReadOnly when a secondary perf cluster node writes to a non-local path, so it can be forwarded to the active primary.
 - The Standby ReadOnly layer, which does indeed sit below barrier and thus below us.

In our case, we _do_ handle the standby read-only explicitly here: https://github.com/hashicorp/vault-enterprise/pull/3427/files#diff-384622a8de9e3bd5d7109571752ed7496b60097a5a4856e945d241a07c1952d6R141-R153 

So I think this comment is just out of date and should be clarified.

Reply:
Ah ok, that explicit check for shouldRequestForward was the missing piece here, thanks for pointing me to it! Given that I think my concern is negated